doc/onboarding_guide.md

```diff
@@ -0,0 +1,95 @@
+Onboarding Best Practices
+=========================
+
+Reach out to `azpycli@microsoft.com` to get the process started. You'll be assigned a dev contact on the CLI team. Early and frequent communication with this contact is essential to ensuring a smooth onboarding.
```

Comment:
Converted this from RST to MD, linked it to the other authoring docs and added more guidance for teams.